sdk/evaluation/azure-ai-evaluation/azure/ai/evaluation/simulator/_adversarial_simulator.py

```diff
@@ -221,6 +221,11 @@ async def __call__(
             random.shuffle(templates)
         parameter_lists = [t.template_parameters for t in templates]
         zipped_parameters = list(zip_longest(*parameter_lists))
+        filtered_parameters = []
```

Comment:
So we basically filter out all of the zipped pairs where one element of the pair is None (aka the shortest parameter list has run out)?

Reply:
yup, if we use `zip` the merge happens on the smallest list and we'll lose some parameters.

Reply:
Doesn't this code have the same effect as `zip` though? If we're making zipped_parameters equal to only the params that don't have a None in the pair (i.e. only ones before the second list runs out).

Let me know if I'm misunderstanding what this chunk does!

Reply:
if the lists being merged are of different lengths, `zip` would merge the lists taking the shortest length and exclude the items beyond that length.

Reply:
`list(zip([1, 2, 3], ['a', 'b']))  # Output: [(1, 'a'), (2, 'b')]`
```list(zip_longest([1, 2, 3], ['a', 'b']))  
# Output: [(1, 'a'), (2, 'b'), (3, None)]```

Reply:
But you might be right. I'll give it a try to use `zip` instead and check if that results in the same output.

Reply:
yup, you're right. `zip` does the job as expected.

Reply:
Awesome!